src/common/Throttle.h

```diff
@@ -332,23 +332,90 @@ class OrderedThrottle {
   uint32_t waiters = 0;
 };
 
+struct Bucket {
+  CephContext *cct;
+  const std::string name;
 
-class TokenBucketThrottle {
+  uint64_t max = 0;
+  uint64_t rate = 0;
```

Comment:
Why a bucket should have a rate? I mean, what is the rate mean in a real Bucket? I don't think this should be in Bucket.

Reply:
In my opinion, the role of the Bucket is speed limit. So, the `rate` is a property of the Bucket.